[CI] Failure in org.elasticsearch.smoketest.XDocsClientYamlTestSuiteIT.test {yaml=en/security/authorization/field-level-security/line_142}

```
/gradlew ':x-pack:docs:integTest' --tests "org.elasticsearch.smoketest.XDocsClientYamlTestSuiteIT.test {yaml=en/security/authorization/field-level-security/line_142}" -Dtests.seed=1029CC6EB70E3475 -Dtests.security.manager=true -Dtests.locale=th -Dtests.timezone=Europe/Jersey -Druntime.java=8 
```


failed with two leaked tasks:

```
  2> java.lang.AssertionError: 2 active tasks found:
    indices:admin/seq_no/global_checkpoint_sync    lq6SkeclQYGplorfF5lpZg:4355 -                           transport 1604488842199 11:20:42 48.9ms      127.0.0.1 node-0 GlobalCheckpointSyncAction.Request{shardId=[ilm-history-3-000001][0], timeout=1m, index='ilm-history-3-000001', waitForActiveShards=DEFAULT}
    indices:admin/seq_no/global_checkpoint_sync[p] lq6SkeclQYGplorfF5lpZg:4356 lq6SkeclQYGplorfF5lpZg:4355 direct    1604488842199 11:20:42 48.6ms      127.0.0.1 node-0 GlobalCheckpointSyncAction.Request{shardId=[ilm-history-3-000001][0], timeout=1m, index='ilm-history-3-000001', waitForActiveShards=1}
     expected:<0> but was:<2>
        at __randomizedtesting.SeedInfo.seed([1029CC6EB70E3475:987DF3B419F2598D]:0)
        at org.junit.Assert.fail(Assert.java:88)
        at org.junit.Assert.failNotEquals(Assert.java:834)
        at org.junit.Assert.assertEquals(Assert.java:645)
        at org.elasticsearch.test.rest.ESRestTestCase.lambda$waitForPendingTasks$3(ESRestTestCase.java:418)
        at org.elasticsearch.test.ESTestCase.assertBusy(ESTestCase.java:1011)
        at org.elasticsearch.test.ESTestCase.assertBusy(ESTestCase.java:984)
        at org.elasticsearch.test.rest.ESRestTestCase.waitForPendingTasks(ESRestTestCase.java:394)
        at org.elasticsearch.xpack.test.rest.AbstractXPackRestTest.cleanup(AbstractXPackRestTest.java:214)
        at sun.reflect.GeneratedMethodAccessor18.invoke(Unknown Source)
        at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
        at java.lang.reflect.Method.invoke(Method.java:498)
        at com.carrotsearch.randomizedtesting.RandomizedRunner.invoke(RandomizedRunner.java:1750)
        at com.carrotsearch.randomizedtesting.RandomizedRunner$10.evaluate(RandomizedRunner.java:996)
        at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
        at org.apache.lucene.util.TestRuleSetupTeardownChained$1.evaluate(TestRuleSetupTeardownChained.java:49)
        at org.apache.lucene.util.AbstractBeforeAfterRule$1.evaluate(AbstractBeforeAfterRule.java:45)
        at org.apache.lucene.util.TestRuleThreadAndTestName$1.evaluate(TestRuleThreadAndTestName.java:48)
        at org.apache.lucene.util.TestRuleIgnoreAfterMaxFailures$1.evaluate(TestRuleIgnoreAfterMaxFailures.java:64)
        at org.apache.lucene.util.TestRuleMarkFailure$1.evaluate(TestRuleMarkFailure.java:47)
        at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
        at com.carrotsearch.randomizedtesting.ThreadLeakControl$StatementRunner.run(ThreadLeakControl.java:368)
        at com.carrotsearch.randomizedtesting.ThreadLeakControl.forkTimeoutingTask(ThreadLeakControl.java:817)
        at com.carrotsearch.randomizedtesting.ThreadLeakControl$3.evaluate(ThreadLeakControl.java:468)
        at com.carrotsearch.randomizedtesting.RandomizedRunner.runSingleTest(RandomizedRunner.java:947)
        at com.carrotsearch.randomizedtesting.RandomizedRunner$5.evaluate(RandomizedRunner.java:832)
        at com.carrotsearch.randomizedtesting.RandomizedRunner$6.evaluate(RandomizedRunner.java:883)
        at com.carrotsearch.randomizedtesting.RandomizedRunner$7.evaluate(RandomizedRunner.java:894)
        at org.apache.lucene.util.AbstractBeforeAfterRule$1.evaluate(AbstractBeforeAfterRule.java:45)
        at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
        at org.apache.lucene.util.TestRuleStoreClassName$1.evaluate(TestRuleStoreClassName.java:41)
        at com.carrotsearch.randomizedtesting.rules.NoShadowingOrOverridesOnMethodsRule$1.evaluate(NoShadowingOrOverridesOnMethodsRule.java:40)
        at com.carrotsearch.randomizedtesting.rules.NoShadowingOrOverridesOnMethodsRule$1.evaluate(NoShadowingOrOverridesOnMethodsRule.java:40)
        at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
        at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
        at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
        at org.apache.lucene.util.TestRuleAssertionsRequired$1.evaluate(TestRuleAssertionsRequired.java:53)
        at org.apache.lucene.util.TestRuleMarkFailure$1.evaluate(TestRuleMarkFailure.java:47)
        at org.apache.lucene.util.TestRuleIgnoreAfterMaxFailures$1.evaluate(TestRuleIgnoreAfterMaxFailures.java:64)
        at org.apache.lucene.util.TestRuleIgnoreTestSuites$1.evaluate(TestRuleIgnoreTestSuites.java:54)
        at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
        at com.carrotsearch.randomizedtesting.ThreadLeakControl$StatementRunner.run(ThreadLeakControl.java:368)
        at java.lang.Thread.run(Thread.java:748)
```

in https://gradle-enterprise.elastic.co/s/upvugf3wfep4w/console-log/raw (7.x)


Does not reproduce locally, but looks like this is just an instance of getting unlucky with the timing of a checkpoint sync and this could happen in other tests as well.

Looking at the proper rending of the exception in Jenkins https://elasticsearch-ci.elastic.co/job/elastic+elasticsearch+7.x+multijob-unix-compatibility/os=oraclelinux-6&&immutable/324/testReport/junit/org.elasticsearch.smoketest/XDocsClientYamlTestSuiteIT/test__yaml_en_security_authorization_field_level_security_line_142_/ one can see that this is just (massive) slowness at the end of the test (but there is movement).

See:

```
	Suppressed: java.lang.AssertionError: 2 active tasks found:
indices:admin/create                        lq6SkeclQYGplorfF5lpZg:4289 -                           transport 1604488831555 11:20:31 738.2ms     127.0.0.1 node-0 
internal:cluster/coordination/publish_state lq6SkeclQYGplorfF5lpZg:4294 -                           direct    1604488831971 11:20:31 322.3ms     127.0.0.1 node-0 
 expected:<0> but was:<2>
		at org.junit.Assert.fail(Assert.java:88)
		at org.junit.Assert.failNotEquals(Assert.java:834)
		at org.junit.Assert.assertEquals(Assert.java:645)
		at org.elasticsearch.test.rest.ESRestTestCase.lambda$waitForPendingTasks$3(ESRestTestCase.java:418)
		at org.elasticsearch.test.ESTestCase.assertBusy(ESTestCase.java:999)
		... 37 more
	Suppressed: java.lang.AssertionError: 2 active tasks found:
indices:admin/create                        lq6SkeclQYGplorfF5lpZg:4289 -                           transport 1604488831555 11:20:31 805.1ms     127.0.0.1 node-0 
internal:cluster/coordination/publish_state lq6SkeclQYGplorfF5lpZg:4294 -                           direct    1604488831971 11:20:31 389.2ms     127.0.0.1 node-0 
 expected:<0> but was:<2>
		at org.junit.Assert.fail(Assert.java:88)
		at org.junit.Assert.failNotEquals(Assert.java:834)
		at org.junit.Assert.assertEquals(Assert.java:645)
		at org.elasticsearch.test.rest.ESRestTestCase.lambda$waitForPendingTasks$3(ESRestTestCase.java:418)
		at org.elasticsearch.test.ESTestCase.assertBusy(ESTestCase.java:999)
		... 37 more
	Suppressed: java.lang.AssertionError: 2 active tasks found:
indices:admin/create                        lq6SkeclQYGplorfF5lpZg:4289 -                           transport 1604488831555 11:20:31 935.6ms     127.0.0.1 node-0 
internal:cluster/coordination/publish_state lq6SkeclQYGplorfF5lpZg:4294 -                           direct    1604488831971 11:20:31 519.6ms     127.0.0.1 node-0 
 expected:<0> but was:<2>
		at org.junit.Assert.fail(Assert.java:88)
		at org.junit.Assert.failNotEquals(Assert.java:834)
		at org.junit.Assert.assertEquals(Assert.java:645)
		at org.elasticsearch.test.rest.ESRestTestCase.lambda$waitForPendingTasks$3(ESRestTestCase.java:418)
		at org.elasticsearch.test.ESTestCase.assertBusy(ESTestCase.java:999)
		... 37 more
	Suppressed: java.lang.AssertionError: 2 active tasks found:
indices:admin/create                        lq6SkeclQYGplorfF5lpZg:4289 -                           transport 1604488831555 11:20:31 1.1s        127.0.0.1 node-0 
internal:cluster/coordination/publish_state lq6SkeclQYGplorfF5lpZg:4294 -                           direct    1604488831971 11:20:31 778ms       127.0.0.1 node-0 
 expected:<0> but was:<2>
		at org.junit.Assert.fail(Assert.java:88)
		at org.junit.Assert.failNotEquals(Assert.java:834)
		at org.junit.Assert.assertEquals(Assert.java:645)
		at org.elasticsearch.test.rest.ESRestTestCase.lambda$waitForPendingTasks$3(ESRestTestCase.java:418)
		at org.elasticsearch.test.ESTestCase.assertBusy(ESTestCase.java:999)
		... 37 more
	Suppressed: java.lang.AssertionError: 2 active tasks found:
indices:admin/create                        lq6SkeclQYGplorfF5lpZg:4289 -                           transport 1604488831555 11:20:31 1.7s        127.0.0.1 node-0 
internal:cluster/coordination/publish_state lq6SkeclQYGplorfF5lpZg:4294 -                           direct    1604488831971 11:20:31 1.2s        127.0.0.1 node-0 
 expected:<0> but was:<2>
		at org.junit.Assert.fail(Assert.java:88)
		at org.junit.Assert.failNotEquals(Assert.java:834)
		at org.junit.Assert.assertEquals(Assert.java:645)
		at org.elasticsearch.test.rest.ESRestTestCase.lambda$waitForPendingTasks$3(ESRestTestCase.java:418)
		at org.elasticsearch.test.ESTestCase.assertBusy(ESTestCase.java:999)
		... 37 more
	Suppressed: java.lang.AssertionError: 1 active tasks found:
indices:admin/create           lq6SkeclQYGplorfF5lpZg:4289 -                           transport 1604488831555 11:20:31 2.7s        127.0.0.1 node-0 
 expected:<0> but was:<1>
		at org.junit.Assert.fail(Assert.java:88)
		at org.junit.Assert.failNotEquals(Assert.java:834)
		at org.junit.Assert.assertEquals(Assert.java:645)
		at org.elasticsearch.test.rest.ESRestTestCase.lambda$waitForPendingTasks$3(ESRestTestCase.java:418)
		at org.elasticsearch.test.ESTestCase.assertBusy(ESTestCase.java:999)
		... 37 more
	Suppressed: java.lang.AssertionError: 2 active tasks found:
indices:admin/create                        lq6SkeclQYGplorfF5lpZg:4289 -                           transport 1604488831555 11:20:31 4.7s        127.0.0.1 node-0 
internal:cluster/coordination/publish_state lq6SkeclQYGplorfF5lpZg:4333 -                           direct    1604488836212 11:20:36 127.8ms     127.0.0.1 node-0 
 expected:<0> but was:<2>
		at org.junit.Assert.fail(Assert.java:88)
		at org.junit.Assert.failNotEquals(Assert.java:834)
		at org.junit.Assert.assertEquals(Assert.java:645)
		at org.elasticsearch.test.rest.ESRestTestCase.lambda$waitForPendingTasks$3(ESRestTestCase.java:418)
		at org.elasticsearch.test.ESTestCase.assertBusy(ESTestCase.java:999)
		... 37 more
```

The timeout here is only 10s, I'll see if there's anything special about this test causing such slowness and if not I'll open a PR to up the timeout.

--- a/test/framework/src/main/java/org/elasticsearch/test/rest/ESRestTestCase.java
+++ b/test/framework/src/main/java/org/elasticsearch/test/rest/ESRestTestCase.java
@@ -395,7 +395,7 @@ public abstract class ESRestTestCase extends ESTestCase {
             } catch (final IOException e) {
                 throw new AssertionError("error getting active tasks list", e);
             }
-        });
+        }, 30L, TimeUnit.SECONDS);
     }
 
     /**

